test/Compressors/Gaussian.jl

```diff
@@ -0,0 +1,157 @@
+            A = rand(4, 2)
+            S_method = complete_compressor(S1, A)
+            @test typeof(S_method) <: CompressorRecipe
+            # Check that default values are correct and appropriate allocations have been made
+            @test S_method.cardinality == Left
+            @test S_method.compression_dim == 2
+            @test S_method.n_rows == S1.compression_dim
+            @test S_method.n_cols == size(A,1)
+            sample_size = S_method.compression_dim
+            @test S_method.scale == 1 / sqrt(sample_size)
+            @test typeof(S_method.op) <: Matrix{eltype(A)}
+            @test size(S_method.op, 2) == size(A, 1)
+            # Test that update_compressor updates the entries
+            sketch_matrix = deepcopy(S_method.op)
+            update_compressor!(S_method)
+            # Check that at least one of the entries is different
+            @test sum(sketch_matrix .== S_method.op) < 4 * 2 
+        end
+
+        # Test the right compressor recipe construction and updating wirh default parameters
+        let
+            seed!(21321)
+            S2 = Gaussian(cardinality = Right)
+            @test typeof(S2) <: Compressor
+            @test S2.cardinality == Right
+            @test S2.compression_dim == 2
+            # Test completion of constructor
+            A = rand(2, 6)
+            S_method = complete_compressor(S2, A)
+            @test typeof(S_method) <: CompressorRecipe
+            # Check that default values are correct and appropriate allocations have been made
+            @test S_method.cardinality == Right
+            @test S_method.compression_dim == 2
+            @test S_method.n_rows == size(A,2)
+            @test S_method.n_cols == S2.compression_dim
+            sample_size = S_method.compression_dim
+            @test S_method.scale == 1 / sqrt(sample_size)
+            @test typeof(S_method.op) <: Matrix{eltype(A)}
+            @test size(S_method.op, 1) == size(A, 2)
+            # Test that update_compressor updates the entries
+            sketch_matrix = deepcopy(S_method.op)
+            update_compressor!(S_method)
+            # Check that at least one of the entries is different
+            @test sum(sketch_matrix .== S_method.op) < 6 * 2 
+        end
+
+        # Test the different multiplications with the left
+        let
+            seed!(2131)
+            n_rows = 10
+            n_cols = 3
+            sketch_size = 6
+            A = rand(n_rows, n_cols)
+            B = rand(sketch_size, n_cols)
+            C1 = rand(sketch_size, n_cols)
+            C2 = rand(n_rows, sketch_size)
+            C3 = rand(n_rows, n_cols)
+            x = rand(n_rows)
+            y = rand(sketch_size)
+            S_info = Gaussian(cardinality = Left, compression_dim = sketch_size)
+            S = complete_compressor(S_info, A)
+            
+            # Form a vector corresponding to the columns to generate the sketch matrix
+            S_test = S.op
```

Comment:
This should be deepcopy? Presumably, S.op is not being changed by any of the multplications.